internal/datasources/network_peers.go

```diff
@@ -0,0 +1,145 @@
+package datasources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	network_peer_api "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/network_peer"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &NetworkPeers{}
+	_ datasource.DataSourceWithConfigure = &NetworkPeers{}
+)
+
+// NetworkPeers is the NetworkPeers data source implementation.
+type NetworkPeers struct {
+	*providerschema.Data
+}
+
+// NewNetworkPeers is a helper function to simplify the provider implementation.
+func NewNetworkPeers() datasource.DataSource {
+	return &NetworkPeers{}
+}
+
+// Metadata returns the network peers data source type name.
+func (n *NetworkPeers) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_network_peers"
+}
+
+// Schema defines the schema for the NetworkPeers data source.
+func (n *NetworkPeers) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = NetworkPeerSchema()
+}
+
+// Read refreshes the Terraform state with the latest data of network peers.
+func (n *NetworkPeers) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state providerschema.NetworkPeers
+	diags := req.Config.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if state.OrganizationId.IsNull() {
```

Comment:
null checks in lines 52-73 probably should be its own validate function

Reply:
Done, thanks!